pkg/resource/endpoint/custom_update_conditions.go

```diff
@@ -51,20 +53,23 @@ func (rm *resourceManager) customUpdateConditions(
 			return false
 		}
 	}
-	if terminalCondition == nil {
-		terminalCondition = &ackv1alpha1.Condition{
-			Type:   ackv1alpha1.ConditionTypeTerminal,
-			Status: corev1.ConditionTrue,
-			Reason: aws.String("Endpoint status: Failed. Cannot be updated"),
-		}
 
-		failureReason := r.ko.Status.FailureReason
-		if failureReason != nil {
-			terminalCondition.Message = failureReason
+	if (err != nil && err == FailUpdateError) || (latestStatus != nil && *latestStatus == svcsdk.EndpointStatusFailed) {
+		// setting terminal condition since controller can no longer recover by retrying
+		if terminalCondition == nil {
+			terminalCondition = &ackv1alpha1.Condition{
+				Type: ackv1alpha1.ConditionTypeTerminal,
+			}
+			ko.Status.Conditions = append(ko.Status.Conditions, terminalCondition)
 		}
```

Comment:
Need an else statement to find the existing terminal condition (if it exists but is set to ConditionFalse)?

Reply:
Good question!
 1. If a terminal condition is already set we do not want to override that, see #51-54
 1. If an error did not occur, this part of code will never execute and the condition will be cleared by [this part of code](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/pkg/resource/endpoint/sdk.go#L343-L346). Hence the else does not exist

Reply:
The terminal condition may be set, but set to False (that case is not covered on lines 51-54)?